File: test/end-to-end/Conversions/Python/converters.py
from math import floor
def try_as_double(doubleLike):
    asDouble = None

    try:
        asDouble = float(doubleLike)
    except:
        pass

    if asDouble is not None:
        print("It's a double: {0}".format(floor(asDouble)))

def try_as_int(firstIntLike, secondIntLike):
    firstInt = None
    secondInt = None

    try:
        firstInt = int(firstIntLike)
        secondInt = int(secondIntLike)
    except:
        pass

    if firstInt is not None and secondInt is not None:
        print("Both are ints: {0} {1}".format(firstInt, secondInt))

File: test/end-to-end/Conversions/Java/Converters.java
package lists;

public static class Converters {
    public static void tryAsDouble(String doubleLike) {
        double asDouble = null;

        try {
            asDouble = Double.parseDouble(doubleLike);
        } catch (NumberFormatException e) { }

        if (asDouble != null) {
            System.out.println(String.format("It's a double: %0$d", (int)Math.floor(asDouble)));
        }
    }

    public static void tryAsInt(String firstIntLike, String secondIntLike) {
        int firstInt = null;
        int secondInt = null;

        try {
            firstInt = Int.parseInt(firstIntLike);
            secondInt = Int.parseInt(secondIntLike);
        } catch (NumberFormatException e) { }

        if (firstInt != null && secondInt != null) {
            System.out.println(String.format("Both are ints: %0$d %1$d", firstInt, secondInt));
        }
    }
}
